### PR TITLE
Added better dimensions handling and resizing, and fixed legend

### PR DESCRIPTION
**Pull request by lorem--ipsum**
Basically allows the chart to resize to its parent container. Thoughts, @chaosmail ? :-)

### CODE REVIEW

File: src/factories/Legend.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ module n3Charts.Factory {
 
       var update = (series) => {
         series
-          .attr('class', (d) => 'item ' + d.getMainType())
+          .attr('class', (d:Utils.SeriesOptions) => 'item ' + d.type.join(' '))
```
Comment (chaosmail) [inline src/factories/Legend.ts]: yeah, sexy

File: src/utils/Dimensions.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module n3Charts.Utils {
+  'use strict';
+
+  export class Dimensions {
```
Comment (chaosmail) [inline src/utils/Dimensions.ts]: Maybe singular would be better: `Dimension`

File: src/utils/Dimensions.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public innerWidth = 560;
+    public innerHeight = 160;
+    public margin = {
+      left: 30,
+      bottom: 20,
+      right: 20,
+      top: 20
+    };
+
+    getPixelCssProp(element: HTMLElement, propertyName: string) {
```
Comment (chaosmail) [inline src/utils/Dimensions.ts]: maybe `getDimensionByProperty`. I like your advice not to use abbreviations

File: src/utils/Dimensions.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      var style = window.getComputedStyle(element, null);
+      return +style.getPropertyValue(propertyName).replace(/px$/, '');
+    }
+
+    fromParentElement(parent: HTMLElement) {
+      // Oooooh I hate doing this.
```
Comment (chaosmail) [inline src/utils/Dimensions.ts]: could we use `box-sizing: border-box;` on the parent, see the part in the bootstrap 4 docs http://v4-alpha.getbootstrap.com/getting-started/introduction/#box-sizing

File: src/utils/Dimensions.ts (hunk 2 above)
Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: Hmm it contains several dimensions, doesn't it ?

File: src/utils/Dimensions.ts (hunk 3 above)
Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: Right, I will fix that

File: src/utils/Dimensions.ts (hunk 4 above)
Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: Hmmmm the parent, like the div generated by the directive ? And I'm not familiar with box-sizing, can you explain a bit more please ? :-)

File: src/utils/Dimensions.ts (hunk 2 above)
Comment (chaosmail) [inline src/utils/Dimensions.ts]: yeah true, only accessing `Dimension.height` would make more sense for me than `Dimensions.height`

File: src/utils/Dimensions.ts (hunk 4 above)
Comment (chaosmail) [inline src/utils/Dimensions.ts]: sure, box sizing is used for the width calculation, and with `border-box` padding is not increasing the width. In the demo you can see, that only the element on the right has width and height of 100px.
Here are the docs: https://developer.mozilla.org/en/docs/Web/CSS/box-sizing
Here is a demo: http://plnkr.co/edit/RfmnhtQUwOaFkCDFHS15?p=preview

Comment (chaosmail) [inline src/utils/Dimensions.ts]: So if we use `box-sizing: border-box;` on the parent, we don't need to subtract the padding ;)

Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: Hm, I can't get it to work... Would you mind trying ?

Comment (chaosmail) [inline src/utils/Dimensions.ts]: no problem, sure. we can merge this and then adapt this

File: src/utils/Dimensions.ts (hunk 2 above)
Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: How about `ChartBoundingBox` ?

Comment (lorem--ipsum) [inline src/utils/Dimensions.ts]: Also, accessing `Dimension.height` looks like accessing `Axis.verticalAxis`, or... I don't know, `Property.individualProperty`. I mean that's a set, or a collection, right ? Height and width are two dimensions, i.e. one is the normal of the other, in my mind, so... Maybe `DimensionSet` ? :-) 

_we'll find a trade-off ^^_

Comment (chaosmail) [inline src/utils/Dimensions.ts]: If I think about it I can live with `Dimensions` ;)

**Comment by chaosmail** [comment]
Are you planning another commit or can I merge this?

**Comment by lorem--ipsum** [comment]
There's still a change I need to push :-)

**Pull request merged**